android/java/org/chromium/chrome/browser/brave_news/CardBuilderFeedCard.java

```diff
@@ -0,0 +1,1421 @@
+import com.bumptech.glide.load.engine.GlideException;
+import com.bumptech.glide.load.resource.bitmap.CenterCrop;
+import com.bumptech.glide.load.resource.bitmap.GranularRoundedCorners;
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
+import com.bumptech.glide.request.RequestListener;
+import com.bumptech.glide.request.RequestOptions;
+import com.bumptech.glide.request.target.Target;
+import android.graphics.drawable.BitmapDrawable;
+import androidx.annotation.Nullable;
+import androidx.annotation.NonNull;
+
+public class CardBuilderFeedCard {
+
+    private final int CARD_LAYOUT = 7;
+    private final int BUTTON_LAYOUT = 8;
+    private final int ROUNDED_TOP_LAYOUT = 9;
+
+    private final int TITLE = 10;
+    private final int DESC = 11;
+    private final int URL = 12;
+    private final int PUBLISHER = 13;
+    private final int TIME = 14;
+    private final int CATEGORY = 15;
+    private final int DEALS = 16;
+
+
+    private LinearLayout linearLayout;
+    private Activity mActivity;
+    private View view;
+    private FeedItemsCard mNewsItem;
+    private int mPosition;
+    private int mType;
+    private int horizontalMargin;
+    private boolean isTablet;
+    private int mDeviceWidth;
+    private boolean isLandscape;
+
+    private String TAG = "BN";
+    private BraveNewsController mBraveNewsController;
+    private boolean isPromo;
+    private String creativeInstanceId;
+    private String offersCategory;
+    private final int MARGIN_VERTICAL = 10;
+    private final String BRAVE_OFFERS_URL = "offers.brave.com";
+
+    public CardBuilderFeedCard(BraveNewsController braveNewsController, LinearLayout layout,
+            Activity activity, int position, FeedItemsCard newsItem, int type) {
+        linearLayout = layout;
+        mActivity = activity;
+        mPosition = position;
+        mType = type;
+        mNewsItem = newsItem;
+        mBraveNewsController = braveNewsController;
+
+        mDeviceWidth = ConfigurationUtils.getDisplayMetrics(activity).get("width");
+
+        isPromo = false;
+        creativeInstanceId = "";
+        offersCategory = "";
+
+        isTablet = ConfigurationUtils.isTablet(
+                activity); // DeviceFormFactor.isNonMultiDisplayContextOnTablet(activity);
+        isLandscape = ConfigurationUtils.isLandscape(activity);
+
+        horizontalMargin = isTablet
+                ? isLandscape ? (int) (0.20 * mDeviceWidth) : (int) (0.10 * mDeviceWidth)
+                : 40;
+        try {
+            createCard(mType, mPosition);
+        } catch (Exception e) {
+            Log.e("bn", "Exception createCard:" + e);
+        }
+    }
+
+    public void initItems() {}
+
+    public void removeCard(LinearLayout layout) {
+        layout.removeAllViews();
+        layout.setVisibility(View.GONE);
+        layout.invalidate();
+    }
+
+    private Url getImage(FeedItemMetadata itemMetaData) {
+        Url imageUrlTemp = null;
```

Comment:
why do we have `imageUrlTemp` and `finalImageUrl`? maybe it would be enough just return whatever we get:

```
        switch(itemMetaData.image.which()) {
            case Image.Tag.PaddedImageUrl:
                return itemMetaData.image.getPaddedImageUrl();
            case Image.Tag.ImageUrl:
                return itemMetaData.image.getImageUrl();
            default:
                return null;

        }
```
?

Reply:
sure. it was easier to log :)